packages/dashboard/src/components/iot-dashboard/iot-dashboard-messages/messages.ts

```diff
@@ -0,0 +1,47 @@
+export type ContextMenuMessages = {
+  copyText: string;
+  pasteText: string;
+  deleteText: string;
+  bringToFrontText: string;
+  sendToBackText: string;
+};
+
+export type WidgetsMessages = {
+  invalidTagHeader: string;
+  invalidTagSubheader: string;
+};
+
+export type DashboardMessages = {
+  widgets: WidgetsMessages;
+  contextMenu: ContextMenuMessages;
+};
+
+export type DashboardMessageOverrides = {
```

Comment:
should this also be a partial?

Reply:
I added the `DashboardMessages` type so that the default messages wouldn't be partial. It made extracting the message easier since we can just do something like -

`Object.assign(
      this.messageOverrides?.contextMenu ?? {},
      DefaultDashboardMessages.contextMenu
    );`
    
    Otherwise we'd have to add another check on the default even though we know it's defined.

Reply:
this forces a user of the dashboard to specify each 'section' even if they don't want to, i.e. i must write:

```
<iot-dashboard messageOverrides={{ widgets: {}, contextMenu: { copyText: 'custom msg' } }} ... />
```

This would also make additional base keys of the message overrides a breaking change.

Reply:
👍 I see your point, updated using RecursivePartial per our discussion!